wagtail/wagtailadmin/messages.py

```diff
@@ -31,5 +33,32 @@ def error(request, message, buttons=None):
     return messages.error(request, render(message, buttons))
 
 
+def validation_error(request, message, form, buttons=None):
+    if not form.non_field_errors():
+        # just output the generic "there were validation errors" message, and leave
+        # the per-field highlighting to do the rest
+        full_message = message
+    else:
+        # display the full list of field and non-field validation errors
+        all_errors = []
+        for field_name, errors in form.errors.items():
+            if field_name == NON_FIELD_ERRORS:
+                prefix = ''
+            else:
+                try:
+                    field_label = form[field_name].label
+                except KeyError:
+                    field_label = field_name
+                prefix = "%s: " % field_label
+
+            for error in errors:
+                all_errors.append(prefix + error)
+
+        errors_html = format_html_join('\n', '<li>{}</li>', ((e,) for e in all_errors))
+        full_message = format_html("""{} <ul class="errorlist">{}</ul>""", message, errors_html)
+
+    return messages.error(request, render(full_message, buttons))
```

Comment:
@jjanssen Good point! While I don't think validation error messages will have buttons in practice, the API allows for that possibility, so it's worth making sure it displays sensibly.

Now fixed in https://github.com/wagtail/wagtail/pull/3440/commits/66c74b7759599d80b2dffffddd069611e3cd9699 by introducing a new 'detail' section to the message template.